src/components/NavBar/index.js

```diff
@@ -0,0 +1,93 @@
+import React, { useState, useEffect, useContext } from 'react'
+import { Link } from 'react-router-dom'
+
+import { UserContext } from '../../store/UserContext'
+
+import AppBar from '@material-ui/core/AppBar'
+import Typography from '@material-ui/core/Typography'
+import Toolbar from '@material-ui/core/Toolbar'
+import Button from '@material-ui/core/Button'
+
+
+
+const NavBar = () => {
+    const user = useContext(UserContext)
+    const [isLoggedIn, setIsLoggedIn] = user.isLoggedIn
+    const [userName, setUserName] = user.userName
+
+    
+    useEffect(() => {
+        console.log("context-user", isLoggedIn)
+        const checkLogin = async () => {
+            // setFetchData(true)
+            let response = await fetch('/users/status')
+            //     .catch(error => {
+            //         console.log("error", error)
+            //         setFetchData(false)
+            //         setFetchError(error)
+            //     })
+            // if (!response.ok) {
+            //     // setFetchData(false)
+            //     console.log(response.statusText)
+            //     return setFetchError(response.statusText)
+            // }
+            // console.log(response)
+            let textResponse = await (response.json())
+            console.log("is logged in ", textResponse)
+            if (textResponse.loggedIn) {
+                setIsLoggedIn(true)
+                setUserName(textResponse.user.userName)
+            }
+            // setFetchData(false)
+        }
```

Comment:
Why do you need to wrap this code in a function?

Reply:
it has to be wrapped in a function to be async.